tiling.py

```diff
@@ -369,22 +405,31 @@ def center(self):
 
 
 class LineKernel(object):
-    def __init__(self, radius, center, group):
+    def __init__(self, radius, vradius, center, group):
         if group.n not in [1]:
             raise ValueError("Cannot construct a line kernel from planar group with dihedral "
                              "subgroup of order not 1")
         self._group = group
         self._radius = radius
         self._diameter = 2 * radius + 1
         self._center = np.array(center)
+        self._vradius = vradius
+        self._vdiameter = 2 * vradius + 1 if self._group.flag_ceiling or self._group.flag_floor else 1
+
 
     def __str__(self):
         return "Line Kernel: %d by %d centered at %s" % (self._diameter, self._diameter, str(self._center))
 
     def adjacent_kernels(self, overlap):
-        for direction in 1, -1:
-            new_ctr = self._center + np.array(direction) * (self._diameter - overlap)
-            yield LineKernel(self._radius, new_ctr, self._group)
+        neighbor_directions = [(1, 0), (-1, 0)]
+        if self._group.flag_ceiling:
+            neighbor_directions.append((0, 1))
+        if self._group.flag_floor:
```

Comment:
This is incorrect. The kernel differs by a translation. If only the floor or only the ceiling are present the only possible translation is along the "tunnel" of the line kernel.

This should be 

```
neighbor_directions = [(1, 0), (-1, 0)]
if self._group.flag_floor and self._group.flag_ceiling:
    neighbor_directions.extend([(0, 1), (0, -1)]

for direction in neighbor_directions:
    new_ctr = self._center + np.array(direction) * (self._diameter - overlap)
    yield LineKernel(self._radius, new_ctr, self._group)
```